flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/triggers/ProcessingTimeoutTrigger.java

```diff
@@ -0,0 +1,139 @@
+	private final boolean resetTimerOnNewRecord;
+	private final boolean shouldClearAtTimeout;
+
+	private final ValueStateDescriptor<Long> timeoutStateDesc;
+
+	private ProcessingTimeoutTrigger(Trigger<T, W> nestedTrigger, long interval, boolean resetTimerOnNewRecord, boolean shouldClearAtTimeout) {
+		this.nestedTrigger = nestedTrigger;
+		this.interval = interval;
+		this.resetTimerOnNewRecord = resetTimerOnNewRecord;
+		this.shouldClearAtTimeout = shouldClearAtTimeout;
+		this.timeoutStateDesc = new ValueStateDescriptor<Long>("timeout", LongSerializer.INSTANCE);
+	}
+
+	@Override
+	public TriggerResult onElement(T element, long timestamp, W window, TriggerContext ctx) throws Exception {
+		TriggerResult triggerResult = this.nestedTrigger.onElement(element, timestamp, window, ctx);
+		if (triggerResult.isFire()) {
+			this.clear(window, ctx);
+			return triggerResult;
+		}
+
+		ValueState<Long> timeoutState = ctx.getPartitionedState(this.timeoutStateDesc);
+		long nextFireTimestamp = ctx.getCurrentProcessingTime() + this.interval;
+		Long timeoutTimestamp = timeoutState.value();
+		if (timeoutTimestamp != null && resetTimerOnNewRecord) {
+			ctx.deleteProcessingTimeTimer(timeoutTimestamp);
+			timeoutState.clear();
+		}
+
+		if (timeoutTimestamp == null) {
+			timeoutState.update(nextFireTimestamp);
+			ctx.registerProcessingTimeTimer(nextFireTimestamp);
+		}
+
+		return triggerResult;
+	}
+
+	@Override
+	public TriggerResult onProcessingTime(long timestamp, W window, TriggerContext ctx) throws Exception {
+		TriggerResult triggerResult = this.nestedTrigger.onProcessingTime(timestamp, window, ctx);
+		if (shouldClearAtTimeout) {
+			this.clear(window, ctx);
+		}
+		return triggerResult.isPurge() ? TriggerResult.FIRE_AND_PURGE : TriggerResult.FIRE;
+	}
+
+	@Override
+	public TriggerResult onEventTime(long timestamp, W window, TriggerContext ctx) throws Exception {
+		this.nestedTrigger.onEventTime(timestamp, window, ctx);
+		return TriggerResult.FIRE;
+	}
+
+	@Override
+	public void clear(W window, TriggerContext ctx) throws Exception {
+		ValueState<Long> timeoutTimestampState = ctx.getPartitionedState(this.timeoutStateDesc);
+		Long timeoutTimestamp = timeoutTimestampState.value();
+		if (timeoutTimestamp != null) {
+			ctx.deleteProcessingTimeTimer(timeoutTimestamp);
+			timeoutTimestampState.clear();
+		}
+		this.nestedTrigger.clear(window, ctx);
+	}
+
+	@Override
+	public String toString() {
+		return "TimeoutTrigger(" + this.nestedTrigger.toString() + ")";
+	}
+
+	/**
+	 * @param nestedTrigger the trigger to apply the {@link ProcessingTimeoutTrigger} on.
+	 * @param interval      the time in milliseconds to apply the timer when element arrive.
+	 * @param <T>           Type of the element.
+	 * @param <W>           Type of the {@link Window}
+	 * @return {@link ProcessingTimeoutTrigger} with the above configuration.
+	 */
+	public static <T, W extends Window> ProcessingTimeoutTrigger<T, W> of(Trigger<T, W> nestedTrigger, long interval) {
```

Comment:
Sorry for the back and forth but please add a more "descriptive" documentation. For example, this is what `CountTrigger` has:
```
Creates a trigger that fires once the number of elements in a pane reaches the given count.
	
@param maxCount The count of elements at which to fire.
@param <W> The type of {@link Window Windows} on which this trigger can operate.
```
I would rather have a good description and have no `@param` descriptions

Reply:
Its okay, better now then never.
So i added more description and added and example, is that enough?